src/tasks/verify-contract-code.ts

```diff
@@ -55,29 +74,42 @@ async function verifyContractCode(
     );
   }
 
-  console.log(
-    `Verifying virtual token contract at address ${virtualToken.address}`,
-  );
-  await verifyContract(ContractName.VirtualToken, virtualToken, hre);
-
-  const realToken = (
-    await hre.ethers.getContractFactory(ContractName.RealToken)
-  )
-    .attach(await virtualToken.cowToken())
-    .connect(hre.ethers.provider);
+  await verifyContract(ContractName.VirtualToken, virtualToken.address, hre);
 
-  console.log(`Verifying real token contract at address ${realToken.address}`);
-  await verifyContract(ContractName.RealToken, realToken, hre);
+  const realTokenAddress = await virtualToken.cowToken();
+  await verifyContract(ContractName.RealToken, realTokenAddress, hre);
 }
 
 async function verifyContract(
   name: ContractName,
-  contract: Contract,
+  address: string,
   hre: HardhatRuntimeEnvironment & { ethers: HardhatEthersHelpers },
 ) {
-  const deployArgs = await (name === ContractName.RealToken
-    ? getDeployArgsFromRealToken
-    : getDeployArgsFromVirtualToken)(contract);
+  console.log(`Verifying contract ${name} at address ${address}`);
+  const contract = (await hre.ethers.getContractFactory(ContractName.RealToken))
```

Comment:
Here, it should be only name, instead of ContractName.RealToken, right?

Reply:
Good catch, fixed it!